sdk/rust/oak/src/grpc/server.rs

```diff
@@ -34,7 +34,10 @@ pub fn init(address: &str) -> Result<Receiver<Invocation>, OakStatus> {
     let (invocation_sender, invocation_receiver) =
         crate::io::channel_create::<Invocation>("gRPC invocation", &Label::public_untrusted())
             .expect("Couldn't create gRPC invocation channel");
-    init_with_sender(address, invocation_sender)?;
+    init_with_sender(address, invocation_sender).map_err(|e| {
+        let _ = invocation_receiver.close();
```

Comment:
It seems a bit odd to rely on `map_err` just for its side effects, and return the input unmodified. I guess an `if let` would be too verbose? Or is there any other pattern that would be suited?

Reply:
Yeah, `map_err` seems off; I'll go for an explicit `match` instead.